cnxarchive/utils/ident_hash.py

```diff
@@ -82,3 +81,186 @@ def join_ident_hash(id, version):
     if version:
         join_args.append(version)
     return HASH_CHAR.join(join_args)
+
+
+class CNXHash(uuid.UUID):
```

Comment:
We could do that. The origenal reason I wanted to make a uuid.UUID subclass was so I could literally overwrite the class function with my own with an import line such as:

``` python
from utils import CNXHash as UUID
```

Then I was just going to invoke the get_shortid() and comparison functions where needed.  But the details of the implementation changed while working with Ross, so I acknowledge  that it doesn't make sense the way it is written now.  Let me see if i can refactor this some and maybe break the class in to separate functions.   The other reason why i grouped all this information in one class is that it's the easiest way to pull up the information that was used to calculate the short_id's from the python debugger.